reladiff/hashdiff_tables.py

```diff
@@ -27,17 +27,23 @@
 logger = logging.getLogger("hashdiff_tables")
 
 
-def diff_sets(a: list, b: list, skip_sort_results: bool) -> Iterator:
-    c = Counter(b)
-    c.subtract(a)
-    x = c.items() if skip_sort_results else sorted(c.items(), key=lambda i: i[0])   # sort by key
-    for k, count in x:
-        if count < 0:
-            sign = "-"
-            count = -count
+def diff_sets(a: list, b: list, skip_sort_results: bool, duplicate_rows_support: bool) -> Iterator:
+    def gen_diff():
+        if duplicate_rows_support:
+            c = Counter(b)
+            c.subtract(a)
+            x = c.items()
+            for k, count in x:
+                if count < 0:
+                    sign = "-"
+                    count = -count
+                else:
+                    sign = "+"
+                yield from repeat((sign, k), count)
```

Comment:
How do you feel about a shorter version?
```suggestion
                yield from (("+", k) if count > 0 else ("-", k) for k, count in c.items() for _ in range(abs(count)))
```

Reply:
I'm okay with it.

My intuition is that repeat is slightly faster, but I'm not sure it matters overall. Also, with this change, I think we can change all "yield from"s to "return".